### PR TITLE
LPS-88714 Client tests for adaptive media in Documents API

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by ruben-pulido** [comment]
ci:test:sf <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ruben-pulido** [comment]
ci:test:relevant <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java [hunk 1]
```diff
@@ -204,6 +213,43 @@ public void testCreateDocumentInFolder() {
 		);
 	}
 
+	@Test
+	public void testCreateImageInDocumentRepository() {
+		_documentHref = ApioClientBuilder.given(
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Are we overring the value _documentHref? in atest

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java [hunk 2]
```diff
@@ -226,6 +272,212 @@ public void testDeleteDocument() {
 		);
 	}
 
+	@Test
+	public void testGetAdaptiveMediaFileFromImageInDocumentRepository()
+		throws IOException {
+
+		String imageFileName = "image.png";
+
+		_documentHref = ApioClientBuilder.given(
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Are we overriding a global variable _documentHref in a test?

Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Exactly!

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java [hunk 3]
```diff
@@ -226,6 +272,212 @@ public void testDeleteDocument() {
...
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_embedded.'Liferay:Document'.find {it.title == '" + imageFileName +
+				"'}._embedded.adaptedMedia._embedded.find {it.resolutionName " +
+					"== 'Thumbnail-300x300'}.contentUrl"
+		);
+
+		URL adaptiveMedia2ContentURL = new URL(adaptiveMedia2ContentHref);
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Do we calculate this URL?

Is something that adaptive media should be checking, right?

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java (hunk 1 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Yes!

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java [hunk 4]
```diff
@@ -367,6 +619,52 @@ public void testGetDocumentsInFolder() {
...
+		).header(
+			"Accept", "application/hal+json"
+		).when(
+		).get(
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_embedded.'Liferay:Document'.find {it.title == '" + imageFileName +
+				"'}.contentUrl"
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Just check the URL is not empty, I think we have a integration tests that check this, right?

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java [hunk 5]
```diff
@@ -226,6 +272,212 @@ public void testDeleteDocument() {
...
+		).post(
+			_documentsHref
+		).then(
+		).extract(
+		).path(
+			"_links.self.href"
+		);
+
+		Awaitility.reset();
+		Awaitility.setDefaultPollInterval(100, TimeUnit.MILLISECONDS);
```
Comment (cgoncas) [inline modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: It is not possible to use awaitility default config?

Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: Yes, we can go with the default config too

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java (hunk 4 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: 👍

File: modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java (hunk 3 above)
Comment (ruben-pulido) [reply modules/apps/headless-apio/media-object/media-object-apio-client-test/src/testIntegration/java/com/liferay/media/object/apio/client/test/MediaObjectApioTest.java]: We do, using addRelativeURL, based on the URI returned from Adaptive Media.

**Comment by ruben-pulido** [comment]
Thanks @cgoncas! Feedback incorporated and PR open to Brian [here](https://github.com/brianchandotcom/liferay-portal/pull/66630).

**Pull request closed without merging**